libs/sdk-backend-mockingbird/src/recordedBackend/catalog.ts

```diff
@@ -131,75 +147,111 @@ class RecordedAvailableCatalogFactory implements IWorkspaceCatalogAvailableItems
         },
     ) {}
 
-    public forDataset(dataset: ObjRef): IWorkspaceCatalogAvailableItemsFactory {
+    public forDataset = (dataset: ObjRef): IWorkspaceCatalogAvailableItemsFactory => {
         return this.withOptions({
             dataset,
         });
-    }
+    };
 
-    public forTypes(types: CatalogItemType[]): IWorkspaceCatalogAvailableItemsFactory {
+    public forTypes = (types: CatalogItemType[]): IWorkspaceCatalogAvailableItemsFactory => {
         return this.withOptions({
             types,
         });
-    }
+    };
 
     // include and exclude tags do not work yet
-    public includeTags(tags: ObjRef[]): IWorkspaceCatalogAvailableItemsFactory {
+    public includeTags = (tags: ObjRef[]): IWorkspaceCatalogAvailableItemsFactory => {
         return this.withOptions({
             includeTags: tags,
         });
-    }
+    };
 
-    public excludeTags(tags: ObjRef[]): IWorkspaceCatalogAvailableItemsFactory {
+    public excludeTags = (tags: ObjRef[]): IWorkspaceCatalogAvailableItemsFactory => {
         return this.withOptions({
             excludeTags: tags,
         });
-    }
+    };
 
-    public withOptions(
+    public withOptions = (
         options: Partial<IWorkspaceCatalogFactoryOptions>,
-    ): IWorkspaceCatalogAvailableItemsFactory {
+    ): IWorkspaceCatalogAvailableItemsFactory => {
         const newOptions = {
             ...this.options,
             ...options,
         };
-        return new RecordedAvailableCatalogFactory(this.workspace, this.groups, this.items, newOptions);
-    }
+        return new RecordedAvailableCatalogFactory(
+            this.workspace,
+            this.config,
+            this.groups,
+            this.items,
+            newOptions,
+        );
+    };
 
-    public forItems(_items: IAttributeOrMeasure[]): IWorkspaceCatalogAvailableItemsFactory {
+    public forItems = (_items: IAttributeOrMeasure[]): IWorkspaceCatalogAvailableItemsFactory => {
         // availability not implemented yet
         return this;
-    }
+    };
 
-    public forInsight(_insight: IInsightDefinition): IWorkspaceCatalogAvailableItemsFactory {
+    public forInsight = (_insight: IInsightDefinition): IWorkspaceCatalogAvailableItemsFactory => {
         // availability not implemented yet
         return this;
-    }
+    };
 
-    public async load(): Promise<IWorkspaceCatalogWithAvailableItems> {
-        return new RecordedAvailableCatalog(this.workspace, this.groups, this.items);
-    }
+    public load = (): Promise<IWorkspaceCatalogWithAvailableItems> => {
+        return Promise.resolve(
```

Comment:
Why the remove of `async` in favor of `Promise.resolve`?

Reply:
fixed